js/dx.aspnet.data.js

```diff
@@ -48,19 +48,22 @@
             if(requiresKey && !keyExpr) {
                 d.reject(new Error("Primary key is not specified (operation: '" + operation + "', url: '" + ajaxSettings.url + "')"));
             } else {
-                ajaxSettings.dataType = "json";
-                if(operation === "load")
+                if(operation === "load") {
                     ajaxSettings.cache = false;
+                    ajaxSettings.dataType = "json";
+                } else {
+                    ajaxSettings.dataType = "text";
+                }
 
                 if(onBeforeSend)
                     onBeforeSend(operation, ajaxSettings);
 
                 $.ajax(ajaxSettings)
-                    .done(function(res) {
+                    .done(function(res, textStatus, xhr) {
                         if(customSuccessHandler)
-                            customSuccessHandler(d, res);
+                            customSuccessHandler(d, res, xhr);
                         else
-                            d.resolve(res);
+                            d.resolve();
```

Comment:
@AlekseyMartynov 

Is this a necessary change?

Reply:
It's intentional.
`CustomStore` doesn't use promise arguments for update & delete operation therefore no need to pass any.

Reply:
I see, thank you for clarifying.